### PR TITLE
[plugin/input/leofs] Modified for LeoFS v1.4

### PR DESCRIPTION
**Pull request by yosukehara**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
Thank you for the pull request @yosukehara.  When adding support for v1.4 we will need to be careful not to break 1.3 support.  Inspecting [this code](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/leofs/leofs.go#L214) it looks like it was not implemented in a fashion that will allow this trivially, and I believe this change will cause problems with backwards compatibility.

I think we may want to switch this plugin to match based on the  OID name instead of index, or maybe we can switch the plugin to use the snmp plugin under the hood.

**Comment by yosukehara** [comment]
@danielnelson Thank you for your comment. I've tested this fix with LeoFS v1.3.8 and v1.4.1-dev. As the result below, there was no issue:

### LeoStorage v1.3.8

```bash
$ ./telegraf --config ./plugins/inputs/leofs/leo_storage.conf --input-filter leofs --test
> leofs,host=storage_0,node=storage_0@127.0.0.1 
  allocated_memory=64522923,
  allocated_memory_5min=64260779,
  ets_memory_usage=4004781,
  ets_memory_usage_5min=4017799,
  mq_num_of_msg_rebalance=0,
  mq_num_of_msg_replication=0,
  mq_num_of_msg_sync_vnode=0,
  num_of_active_objects=68,
  num_of_deletes=0,
  num_of_deletes_5min=0,
  num_of_processes=580,
  num_of_processes_5min=580,
  num_of_reads=0,
  num_of_reads_5min=0,
  num_of_writes=0,
  num_of_writes_5min=0,
  processes_memory_usage=20461187,
  processes_memory_usage_5min=20561825,
  system_memory_usage=26198287,
  system_memory_usage_5min=26206637,
  total_memory_usage=46658415,
  total_memory_usage_5min=46764434,
  total_objects=69,
  total_size=2,
  total_size_of_active_objects=2,
  used_allocated_memory=69,
  used_allocated_memory_5min=69
  1524445349000000000
```

### LeoStorage v1.4.0-dev

```
$ ./telegraf --config ./plugins/inputs/leofs/leo_storage.conf --input-filter leofs --test
> leofs,host=storage_0,node=storage_0@127.0.0.1
  allocated_memory=0,
  allocated_memory_5min=0,
  comp_last_end_datetime=0,
  comp_last_start_datetime=0,
  comp_num_of_ongoing_targets=0,
  comp_num_of_out_of_targets=0,
  comp_num_of_pending_targets=0,
  comp_state=0,
  ets_memory_usage=0,
  ets_memory_usage_5min=0,
  mq_mdcr_num_of_msg_req_comp_metadata=0,
  mq_mdcr_num_of_msg_req_sync_obj=0,
  mq_num_of_msg_async_deletion_dir=0,
  mq_num_of_msg_deletion_dir=0,
  mq_num_of_msg_rebalance=0,
  mq_num_of_msg_recovery_node=0,
  mq_num_of_msg_replication=0,
  mq_num_of_msg_req_deletion_dir=0,
  mq_num_of_msg_sync_vnode=0,
  num_of_active_objects=0,
  num_of_deletes=0,
  num_of_deletes_5min=0,
  num_of_processes=0,
  num_of_processes_5min=0,
  num_of_reads=0,
  num_of_reads_5min=0,
  num_of_writes=0,
  num_of_writes_5min=0,
  processes_memory_usage=0,
  processes_memory_usage_5min=0,
  system_memory_usage=0,
  system_memory_usage_5min=0,
  total_memory_usage=0,
  total_memory_usage_5min=0,
  total_objects=0,
  total_size=0,
  total_size_of_active_objects=0,
  used_allocated_memory=0,
  used_allocated_memory_5min=0
  1524446576000000000
```
* [leo_storage.conf](https://gist.github.com/yosukehara/c923ebc2ac37920cea937663229efd17)

File: plugins/inputs/leofs/leofs.go [hunk 1]
```diff
@@ -86,13 +86,26 @@ var KeyMapping = map[ServerType][]string{
 		"total_objects",
 		"total_size_of_active_objects",
 		"total_size",
-		"num_of_replication_messages",
-		"num_of_sync-vnode_messages",
-		"num_of_rebalance_messages",
+		"mq_num_of_msg_replication",
+		"mq_num_of_msg_sync_vnode",
+		"mq_num_of_msg_rebalance",
```
Comment (danielnelson) [inline plugins/inputs/leofs/leofs.go]: We need to stay with the original names, or we could report it under both names if you think this is an important rename.

**Comment by danielnelson** [comment]
Thanks!

**Pull request merged**